File: examples/javascript/route.js
import { OSRMPlus } from "osrmplus";

const client = new OSRMPlus("your_api_key");

const result = await client.route([
  [74.3436, 31.5497], // Lahore, start
  [74.265, 31.47],    // Lahore, end
]);

const leg = result.routes[0];
console.log(`Distance: ${leg.distance.toFixed(0)} meters`);
console.log(`Duration: ${leg.duration.toFixed(0)} seconds`);

File: examples/python/route.py
"""Get a route between two points."""

from osrmplus import Client

client = Client("your_api_key")

route = client.route([
    (74.3436, 31.5497),  # Lahore, start
    (74.2650, 31.4700),  # Lahore, end
])

leg = route["routes"][0]
print(f"Distance: {leg['distance']:.0f} meters")
print(f"Duration: {leg['duration']:.0f} seconds")
